pkg/apis/edge/v2alpha1/edge-placementdecision.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SourceEdgePlacementLabelKey is the key of the label used in a EdgePlacementDecision.
+// to reference the EdgePlacement that it is part of the response to.
+// We use a label rather than a field because field selectors do not work
+// on a resource defined by a CRD.
+const SourceEdgePlacementLabelKey string = "edge.kubestellar.io/source-edgeplacement"
+
+// EdgePlacementDecision exists in the center and is bound to a single EdgePlacement resource.
+// The decision resource reflects the resolution of the bounded EdgePlacement's placement selectors,
+// and explicitly reflects which resources should go to what destinations.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster,shortName=epl
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type EdgePlacementDecision struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// `spec` explicitly describes a desired binding between workloads and Locations.
+	// It reflects the resolution of an EdgePlacement's placement selectors.
+	// +optional
+	Spec EdgePlacementDecisionSpec `json:"spec,omitempty"`
+
+	// `status` describes the status of the process of binding workloads to Locations.
+	// +optional
+	Status EdgePlacementDecisionStatus `json:"status,omitempty"`
+}
+
+// EdgePlacementDecisionSpec holds a list of resources and a list of destinations which are the resolution
+// of an EdgePlacement's `what` and `where`: what resources to propagate and to where.
+// All resources present in this spec are propagated to all destinations present.
+type EdgePlacementDecisionSpec struct {
+	// `Workload` is a collection of namespaced and cluster-scoped resources to be propagated to destination clusters.
+	Workload EdgePlacementDecisionSpecResources `json:"workload,omitempty"`
+
+	// `destinations` is a list of cluster-identifiers that the resources should be propagated to.
+	Destinations []Destination `json:"destinations,omitempty"`
+}
+
+// EdgePlacementDecisionSpecResources explicitly defines the resources to be synced.
```

Comment:
fix comment - "to be downsynced". people who are familiar with current upsync/downsync capabilites might get confused.